tests/integration/backend/filestate/backend_gcp_test.go

```diff
@@ -0,0 +1,22 @@
+package filestate
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+//nolint:paralleltest // this test sets the global login state
+func TestGcpLogin(t *testing.T) {
+	err := os.Chdir("project")
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		err := os.Chdir("..")
+		require.NoError(t, err)
+	})
+
+	t.Setenv("GOOGLE_PROJECT", "pulumi-ci-gcp-provider")
```

Comment:
This should probably have a skip case for when whatever the gcp auth var isn't set.